core/src/main/java/de/danoeh/antennapod/core/service/download/AntennapodHttpClient.java

```diff
@@ -260,7 +286,15 @@ public Socket createSocket(InetAddress var1, int var2, InetAddress var3, int var
         }
 
         private void configureSocket(SSLSocket s) {
-            s.setEnabledProtocols(new String[] { "TLSv1.2", "TLSv1.1", "TLSv1" } );
+            if (Flavors.FLAVOR == Flavors.FREE) {
+                // Free flavor (bundles modern conscrypt): TLSv1.3 and modern cipher suites are
+                // guaranteed. Protocols older than TLSv1.2 are now deprecated and can be disabled.
+                s.setEnabledProtocols(new String[] { "TLSv1.3", "TLSv1.2" });
```

Comment:
Are you sure that this does not break anything? Podcast publishers use pretty horrible setups sometimes (invalid xml, broken certificates, duplicate or changing GUIDs, old tls versions (?)) and users then blame AntennaPod for that. What are the default protocols if we did not call this at all?

Reply:
As long as
```
sslContext = SSLContext.getInstance("TLSv1.3");
```
gets run, TLSv1.3 and TLSv1.2 should be supported on all devices without setEnabledProtocols()... I think. My experiments so far indicates this (on the devices I've tested) but I can't guarantee it. What other protocols are enabled seems to vary by platform and doesn't have a specific default(?). I think SSL3 gets enabled by default on older devices unless explicitly disabled (like this).

Apart form just ensuring that TLSv1.3 and TLSv1.2 are enabled (inspired by the original reason for this class), my main reason for this was to ensure old protocols are not enabled. This includes SSL3, TLS1 and TLS1.1.

I understand your concern. If there is a chance a podcast out there still doesn't support TLSv1.2, this could be a problem. But I've read Firefox, Chrome/Chromium and Safari have or are dropping support for TLS1.1 (and older) this year. But if that's not enough to force server administrators to support modern protocols, it makes me think that maybe we should re-enable SSLv3? I mean, maybe there's a podcast server out there that doesn't even support TLS at all?

Do you know some way to try finding podcasts that might not support TLSv1.2? Obviously the ones I'm listening to works, but that's an insignificant subset of all podcasts out there...

Reply:
> Do you know some way to try finding podcasts that might not support TLSv1.2?

Not really :/ One could ask the gpodder developers: They crawl podcasts anyway. They have not been too active in the last few months and I do not want to cause them additional work, though.

> I've read Firefox, Chrome/Chromium and Safari have or are dropping support for TLS1.1 (and older) this year

Okay, let's remove it and see if users complain.